scantool/scantool_850.c

```diff
@@ -95,13 +95,454 @@ struct dtc_table_entry {
+	{0x11, 0xFE, 131, "Engine Speed (RPM) signal"},
+	//---------------------------------------------------------
+	//  Define all known, interpreted ECU 41 (Immobilizer) DTCs.
+	//---------------------------------------------------------
+	// ECU 41 has 23 DTCs: 01 02 03 04 05 06 07 30 31 32 33 34 35 36 37 38 51 52 53 54 55 F7 F8
+	{0x41, 0x01, 112, "Internal EEPROM memory fault"},
+	{0x41, 0x02, 211, "Comm fault with engine ECU. [might be either IMM-211 or IMM-213]"},
+	{0x41, 0x03, 212, "No Contact with Antenna. [might be either IMM-212 or IMM-233]"},
+	{0x41, 0x04, 221, "No Response from Key Transponder. [might be either IMM-221 or IMM-234]"},
+	{0x41, 0x05, 222, "Key Code not in memory. [might be either IMM-222 or IMM-235]"},
+	{0x41, 0x06, 223, "Comm fault from Transponder to IMMO Control Module"},
+	{0x41, 0x07, 336, "Control circuit to VGLA, fault in VGLA or LED circuit (P600)"},
+	{0x41, 0x30, 334, "Control circuit to Indicator Lamp, fault in circuit. [might be either IMM-334 or IMM-331]"},
+	{0x41, 0x31, 335, "Starter Motor Control circuit, fault in circuit. [might be either IMM-335 or IMM-332]"},
+	{0x41, 0x32, 324, "VERLOG code missing"},
+	{0x41, 0x33, 325, "Comm fault with coded starter module (CSM). [might be either IMM-325 or IMM-323]"},
+	{0x41, 0x34, 121, "PIN programming failed"},
+	{0x41, 0x35, 225, "Key Code verification failed"},
+	{0x41, 0x36, 132, "No Key Codes in EEPROM"},
+	{0x41, 0x37, 122, "New replacement Immobilizer not yet programmed"},
+	{0x41, 0x38, 999, "?EEPROM error?"},
+	{0x41, 0x51, 311, "Comms with engine ECU, short to supply in comm link 2 or 4"},
+	{0x41, 0x52, 312, "Comms with engine ECU, short to ground in comm link 2 or 4"},
+	{0x41, 0x53, 214, "Comms from engine ECU, wrong MIN code (ie, Immobilizer is programmed for a different engine ECU)"},
+	{0x41, 0x54, 321, "Initiating signal from ECM, missing"},
+	{0x41, 0x55, 326, "Reply signal from engine control module"},
+	{0x41, 0xF7, 333, "Control circuit to VGLA, fault in VGLA or LED circuit"},
+	{0x41, 0xF8, 336, "Control circuit to VGLA, fault in VGLA or LED circuit (P600)"},
+	//-------------------------------------------------------------------------------
+	//  Define all known, interpreted ECU 29 (ECC) [for '98 S70/V70/XC70 only] DTCs.
+	//  * There are many more ECC DTCs whose meaning is not yet known.
+	//-------------------------------------------------------------------------------
+	// ECU 29 has 4 DTCs: 35 36 37 38
+	{0x29, 0x35, 412, "Driver's (or passenger's) side interior temperature sensor inlet fan shorted to earth (or signal too low) [see \"ac heater system auto.pdf\"]"},
+	{0x29, 0x36, 413, "Driver's (or passenger's) side interior temperature sensor inlet fan, no control voltage (or signal too high) [see \"ac heater system auto.pdf\"]"},
+	{0x29, 0x37, 414, "Driver's (or passenger's) side interior temperature sensor inlet fan seized (or open/short circuit or faulty signal) [carefully clean out any lint from temp sensor fan(s)]"},
```

Comment:
I guess these say "Driver's (or passenger's)" because which side is the driver's side depends on whether it's a left or right hand drive vehicle. But VADIS says 4-1-2 is for the driver's side sensor intake fan whether you have a LHD or RHD vehicle selected. So either the localization in VADIS is wrong, or the driver's side DTCs are driver's side regardless of market.

Reply:
Looking into this a bit further... up to model year '97 there were two passenger compartment temperature sensors, in the headliner, one on each side. From model year '98 up, there was a single passenger compartment temperature sensor, between the steering wheel and the climate controls. '98 was the first model year where the ECC could be queried by D2; before that, it could only do blink codes, so we can't talk to it. (Although it would be cool to try to get blink code reading working with a KKL cable and the appropriate wiring!)

So the only ECC we can talk to uses a single interior temperature sensor... we should just drop "driver's side" from these DTCs.